osu.Game.Rulesets.Mania/Difficulty/Skills/Strain.cs

```diff
@@ -4,98 +4,281 @@
+        private double globalStrain;
 
         public Strain(Mod[] mods, int totalColumns)
             : base(mods)
         {
-            startTimes = new double[totalColumns];
-            endTimes = new double[totalColumns];
-            individualStrains = new double[totalColumns];
-            overallStrain = 1;
+            prevStartTimes = new double[totalColumns];
+            prevEndTimes = new double[totalColumns];
+            columnStrains = new double[totalColumns];
+            globalStrain = 1;
         }
 
+        /// <summary>
+        /// Calculates the strain value of a <see cref="DifficultyHitObject"/>. This value is affected by previously processed objects.
+        ///
+        /// The function documentation heavily references diagrams the top of the file.
+        ///
+        /// <param name="current">Current Hit Object to evaluate strain</param>
+        /// <remarks>
+        /// 1) The first hitObject is not considered in the calculation.
+        /// <see cref="ManiaDifficultyCalculator.CreateDifficultyHitObjects"/>
+        /// 2) Strain is binned/bucketed by sections by Section Length, with a Max aggregation
+        /// <see cref="StrainSkill.SectionLength"/>
+        /// <see cref="StrainSkill.Process"/>
+        /// </remarks>
+        /// </summary>
         protected override double StrainValueOf(DifficultyHitObject current)
         {
-            var maniaCurrent = (ManiaDifficultyHitObject)current;
-            double startTime = maniaCurrent.StartTime;
-            double endTime = maniaCurrent.EndTime;
-            int column = maniaCurrent.BaseObject.Column;
-            bool isOverlapping = false;
+            var hitObject = (ManiaDifficultyHitObject)current;
 
-            double closestEndTime = Math.Abs(endTime - startTime); // Lowest value we can assume with the current information
-            double holdFactor = 1.0; // Factor to all additional strains in case something else is held
-            double holdAddition = 0; // Addition to the current note in case it's a hold and has to be released awkwardly
+            // Given a note, startTime == endTime.
+            double startTime = hitObject.StartTime;
+            double endTime = hitObject.EndTime;
+            int column = hitObject.BaseObject.Column;
+            double holdLength = Math.Abs(endTime - startTime);
 
-            for (int i = 0; i < endTimes.Length; ++i)
+            // See top of file for interpretation
+            double endOnBodyBias = 0; // Strain Bias for Column 3 states
+            double endAfterTailWeight = 1.0; // Strain Weight for Column 5 states
+            bool isEndOnBody = false; // Flag for Column 3 states
+            bool isEndAfterTail = false; // Flag for Column 5 states
+
+            // The closest end time, currently, is the current note's end time, which is its length
+            // This is used for endOnBodyBias calculation.
+            double closestEndTime = holdLength;
```

Comment:
This is supremely confusing. `holdLength` the length of the hold. It is not the current note's end time, as the current note's end time is `endTime` and as such an absolute value.

The actual piece of important information here is that `closestEndTime` is _relative_ to `startTime`.

Reply:
Good catch, I'll see how to rephrase this